tests/client_api.cpp

```diff
@@ -1561,3 +1561,52 @@ TEST(Groups, Profiles)
 
         alice->close();
 }
+
+TEST(ClientAPI, PublicRooms)  
+{
+        std::shared_ptr<Client> mtx_client = std::make_shared<Client>("localhost");
+        mtx_client->login("alice", "secret", [mtx_client](const mtx::responses::Login &, RequestErr err) {
+                check_error(err);
+        });
+
+        while (mtx_client->access_token().empty()) {
+                sleep();
+        }
+
+        // create a new public room
+        mtx::requests::CreateRoom req;
+        req.name = "Public Room";
+        req.topic = "Test";
+        req.visibility = Visibility::Public;
```

Comment:
That does not do, what you are expecting, I think. You need to send a request to `/_matrix/client/r0/directory/list/room/{roomId}`, I think.